frontend/src/slices/gameDataSlice.ts

```diff
@@ -34,6 +34,11 @@ export const gameDataSlice = createSlice({
       state.currencyAmount = action.payload.currencyAmount;
       state.numBits = action.payload.numBits;
       state.upgrades = action.payload.upgrades;
+    },
+    resetGameData: (state) => {
```

Comment:
I see we are manually resetting the state back to initial conditions. I think I remember you saying that you couldn't use `initialState` for some reason. Could you explain why that didn't work?

Reply:
Also curious about this - if there's a way to do like `state = initialState` or something, that would make it much more sustainable if we add additional game data that has to be reset.

Reply:
This was changed with the new arrow function solution.